src/main/webapp/app/atlas/agent/src/main.py

```
@@ -0,0 +1,26 @@
+    message: str
+
+@app.post("/chat")
+async def chat_endpoint(req: ChatRequest):
+        try:
+         logger.info(f"Received chat request: {req.message[:50]}...")
+         # If handle_prompt is CPU-intensive, consider running in thread pool
+         reply = agent.handle_prompt(req.message)
+         return {"reply": reply}
+        except Exception as e:
+         logger.error(f"Error processing chat request: {e}")
+         raise HTTPException(status_code=500, detail="Internal server error")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)